sapmexporter crashing the otel-collector when passing the signalfx access token

**Describe the bug**

When passing the `com.splunk.signalfx.access_token` using either the splunk-otel-java or official agent the sapm exporter crashes the collector.

**Steps to reproduce**

I built this package https://github.com/spring-projects/spring-petclinic and use the splunk-otel-java and also https://github.com/open-telemetry/opentelemetry-java-instrumentation both produces the same result

**What did you expect to see?**

Traces successfully sent to signalfx.

**What did you see instead?**

```
panic: runtime error: index out of range [13] with length 13

goroutine 131 [running]:
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sapmexporter.filterTokenFromProcess(0xc033f3e820)
	/home/circleci/project/exporter/sapmexporter/exporter.go:157 +0x294
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sapmexporter.filterToken(0xc0005ea560, 0x1, 0x1)
	/home/circleci/project/exporter/sapmexporter/exporter.go:148 +0x3c
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sapmexporter.(*sapmExporter).pushTraceData(0xc00076aae0, 0x42ab700, 0xc00011b680, 0xc000114600, 0x3bb0ae0, 0xc0000ad6b8, 0x49978b)
	/home/circleci/project/exporter/sapmexporter/exporter.go:119 +0xcd
go.opentelemetry.io/collector/exporter/exporterhelper.(*tracesRequest).export(0xc033f1c220, 0x42ab700, 0xc00011b680, 0x42ab700, 0xc00011b680, 0xc000830ad0)
	/home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.22.0/exporter/exporterhelper/tracehelper.go:53 +0x47
go.opentelemetry.io/collector/exporter/exporterhelper.(*timeoutSender).send(0xc0007448b0, 0x42c3140, 0xc033f1c220, 0x0, 0x0, 0x0)
	/home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.22.0/exporter/exporterhelper/common.go:217 +0x92
go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send(0xc0006d7da0, 0x42c3140, 0xc033f1c220, 0x3a5d357, 0x7, 0x42ab740)
	/home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.22.0/exporter/exporterhelper/queued_retry.go:232 +0x281
go.opentelemetry.io/collector/exporter/exporterhelper.(*tracesExporterWithObservability).send(0xc00076ae00, 0x42c3140, 0xc033f1c220, 0x42c3140, 0xc033f1c220, 0x0)
	/home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.22.0/exporter/exporterhelper/tracehelper.go:114 +0xcd
go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).start.func1(0x3584860, 0xc033f1c220)
	/home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.22.0/exporter/exporterhelper/queued_retry.go:133 +0x75
github.com/jaegertracing/jaeger/pkg/queue.ConsumerFunc.Consume(0xc0007802f0, 0x3584860, 0xc033f1c220)
	/home/circleci/go/pkg/mod/github.com/jaegertracing/jaeger@v1.22.0/pkg/queue/bounded_queue.go:104 +0x3a
github.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumersWithFactory.func1(0xc000744920, 0xc0007840f0)
	/home/circleci/go/pkg/mod/github.com/jaegertracing/jaeger@v1.22.0/pkg/queue/bounded_queue.go:83 +0xe9
created by github.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumersWithFactory
	/home/circleci/go/pkg/mod/github.com/jaegertracing/jaeger@v1.22.0/pkg/queue/bounded_queue.go:73 +0xc0
```

**What version did you use?**

Collector 0.22.0
Java agent both 0.10.0 and 0.9.0

--- a/exporter/sapmexporter/exporter.go
+++ b/exporter/sapmexporter/exporter.go
@@ -153,14 +153,13 @@ func filterTokenFromProcess(proc *model.Process) {
 	if proc == nil {
 		return
 	}
-	for i := range proc.Tags {
+	for i := 0; i < len(proc.Tags); {
 		if proc.Tags[i].Key == splunk.SFxAccessTokenLabel {
-			// Switch this tag with last one.
-			lastPos := len(proc.Tags) - 1
-			tmp := proc.Tags[lastPos]
-			proc.Tags[lastPos] = proc.Tags[i]
-			proc.Tags[i] = tmp
-			proc.Tags = proc.Tags[0:lastPos]
+			proc.Tags[i] = proc.Tags[len(proc.Tags)-1]
+			// We do not need to put proc.Tags[i] at the end, as it will be discarded anyway
+			proc.Tags = proc.Tags[:len(proc.Tags)-1]
+			continue
 		}
+		i++
 	}
 }

--- a/exporter/sapmexporter/exporter_test.go
+++ b/exporter/sapmexporter/exporter_test.go
@@ -72,9 +72,15 @@ func buildTestTraces(setTokenLabel bool) (traces pdata.Traces) {
 	for i := 0; i < 20; i++ {
 		rs := rss.At(i)
 		resource := rs.Resource()
+		resource.Attributes().InsertString("key1", "value1")
 		if setTokenLabel && i%2 == 1 {
 			tokenLabel := fmt.Sprintf("MyToken%d", i/5)
 			resource.Attributes().InsertString("com.splunk.signalfx.access_token", tokenLabel)
+			resource.Attributes().InsertString("com.splunk.signalfx.access_token", tokenLabel)
+		}
+		// Add one last element every 3rd resource, this way we have cases with token last or not.
+		if i%3 == 1 {
+			resource.Attributes().InsertString("key2", "value2")
 		}
 
 		span := rs.InstrumentationLibrarySpans().AppendEmpty().Spans().AppendEmpty()

